src/CMakeLists.txt

```diff
@@ -32,39 +37,61 @@ set(CUDASRC
   cuda/sendrecv.f90
   cuda/tdsops.f90
 )
+set(2DECOMPFFTSRC
+  omp/poisson_fft.f90
+  decomp/decomp_2decompfft.f90
+)
+set(DUMMYDECOMPSRC
+  decomp/decomp_dummy.f90
+)
 
-if(${CMAKE_Fortran_COMPILER_ID} STREQUAL "PGI" OR
-   ${CMAKE_Fortran_COMPILER_ID} STREQUAL "NVHPC")
+if(${BACKEND} STREQUAL "CUDA")
   list(APPEND SRC ${CUDASRC})
 endif()
 
+# Decide whether 2decomp&fft is supported by the build
+find_package(decomp2d)
+set(WITH_2DECOMPFFT (${POISSON_SOLVER} STREQUAL "FFT" AND decomp2d_FOUND))
+
+if(WITH_2DECOMPFFT)
+  list(APPEND SRC ${2DECOMPFFTSRC})
+else()
+  list(APPEND SRC ${DUMMYDECOMPSRC})
+endif()
+
 add_library(x3d2 STATIC ${SRC})
 target_include_directories(x3d2 INTERFACE ${CMAKE_CURRENT_BINARY_DIR})
 
 add_executable(xcompact xcompact.f90)
 target_link_libraries(xcompact PRIVATE x3d2)
 
-if(${CMAKE_Fortran_COMPILER_ID} STREQUAL "PGI" OR
-   ${CMAKE_Fortran_COMPILER_ID} STREQUAL "NVHPC")
- 
+if(${COMPILER} STREQUAL "CUDA")
   set(CMAKE_Fortran_FLAGS "-cpp -cuda")
   set(CMAKE_Fortran_FLAGS_DEBUG "-g -O0 -traceback -Mbounds -Mchkptr -Ktrap=fp")
   set(CMAKE_Fortran_FLAGS_RELEASE "-O3 -fast")
   target_link_options(x3d2 INTERFACE "-cuda")
   target_link_options(x3d2 INTERFACE "-cudalib=cufftmp")
 
   target_compile_options(xcompact PRIVATE "-DCUDA")
-elseif(${CMAKE_Fortran_COMPILER_ID} STREQUAL "GNU")
+elseif(${COMPILER} STREQUAL "GNU")
```

Comment:
Why this change? It's probably better to stick to the CMAKE variables

Reply:
if this is a pain to fix we could lift it into a separate issue.

Reply:
The "COMPILER" is directly based on the CMAKE variable, I found it cleaner to be consistent with cuda. But happy to switch back, it turns out it is used only in this if statement, so making an extra variable isn't as needed as it used to.

fyi, the main cmakelists.txt has:

```cmake
if(${CMAKE_Fortran_COMPILER_ID} STREQUAL "PGI" OR
   ${CMAKE_Fortran_COMPILER_ID} STREQUAL "NVHPC")
   set(COMPILER CUDA)
elseif(${CMAKE_Fortran_COMPILER_ID} STREQUAL "GNU")
   set(COMPILER GNU)
else()
   set(COMPILER OTHER)
endif()
```

Reply:
ok, then we leave this to be consistent with the rest of the code.

Reply:
Just changed it back, I think you are right, it is better to not use 'custom' variables when it isn't really needed. And because it is used only here there is no point in having it. It was more useful before when the compiler and backend were the same thing, but because that isn't the case anymore, it isn't as needed.